weblate/trans/models/component.py

```diff
@@ -1179,6 +1211,95 @@ def get_repoweb_link(
             component=self,
         )
 
+    def get_git_repoweb_template(self):
+        """Method to return the template link for a specific vcs."""
+        parsed_url = urlparse(self.repo)
+
+        if (
+            self.repo.startswith("git@bitbucket.org")
+            or parsed_url.hostname == "bitbucket.org"
+        ):
+            return self.get_bitbucket_git_repoweb_template()
+
+        if (
+            self.repo.startswith("git@github.com")
+            or parsed_url.hostname == "github.com"
+        ):
+            return self.get_github_repoweb_template()
+
+        if "pagure.io" in self.repo:
+            return self.get_pagure_repoweb_template()
+
+        if (
+            self.repo.startswith("git@ssh.dev.azure.com:v3")
+            or parsed_url.hostname == "dev.azure.com"
+            or "visualstudio.com" in self.repo
+        ):
+            return self.get_azure_repoweb_template()
+
+        return None
+
+    def get_bitbucket_git_repoweb_template(self):
+        owner, slug, matches = None, None, None
+        if re.match(BITBUCKET_GIT_REPOS_REGEXP[0], self.repo):
+            matches = re.search(BITBUCKET_GIT_REPOS_REGEXP[0], self.repo)
+        elif re.match(BITBUCKET_GIT_REPOS_REGEXP[1], self.repo):
+            matches = re.search(BITBUCKET_GIT_REPOS_REGEXP[1], self.repo)
+        if matches:
+            owner = matches.group(1)
+            slug = matches.group(2)
+        if owner and slug:
+            return f"https://bitbucket.org/{owner}/{slug}"
+
+        return None
+
+    def get_github_repoweb_template(self):
+        owner, slug, matches = None, None, None
+        if re.match(GITHUB_REPOS_REGEXP[0], self.repo):
+            matches = re.search(GITHUB_REPOS_REGEXP[0], self.repo)
+        elif re.match(GITHUB_REPOS_REGEXP[1], self.repo):
+            matches = re.search(GITHUB_REPOS_REGEXP[1], self.repo)
+        if matches:
+            owner = matches.group(1)
+            slug = matches.group(2)
+        if owner and slug:
+            return f"https://github.com/{owner}/{slug}"
```

Comment:
What we really need here is:

```suggestion
            return f"https://{domain}/{owner}/{slug}/blob/{{branch}}/{{filename}}#L{{line}}"
```